### PR TITLE
[ENH] Made UI responsive for smaller screens

### LINKED ISSUE #122: Make UI responsive for smaller screens
# Context
Right now the tool works best on desktop screen sizes. On smaller screens it scales down until Ui elements overflow their boxes - and look bad.

Because we use tailwind it should be easy to stack the query form on top of the response container when the screen size is smaller. That should already help.

# Why
- right now the app looks not very nice outside of a certain range of (pretty wide) screen width
- have a good basis to keep the app responsive (i.e. in the future we don't want to have responsiveness issues come back)
- have all UI elements be reasonably spaced apart so they don't overlap / touch

# Outcomes

- [x] If I make the screen very narrow, UI elements do not overlap
- [x] UI elements go from double-column layout to a narrow-screen appropriate layout
- [x] when I navigate to the query tool with a mobile-sized screen, I get some clear indication that the screen size is not supported (we probably want to figure out what screen sizes we want to / should support)

# What

- [ ]

### LINKED ISSUE #333: Read about changes required to make UI responsive
The goal is to find tools/ways/best practices to modify the existing UI so that elements are reasonably spaced and do not overlap when screen size changes.

A good starting point for research would be to take a look at the material design docs on responsive layout: https://m2.material.io/design/layout/responsive-layout-grid.html#columns-gutters-and-margins

- [ ] ~~summarize strategies for responsiveness~~
- [ ] update https://github.com/neurobagel/query-tool/issues/122 with proposed changes
- [x] (optional) add interesting learnings in appropriate section of https://github.com/neurobagel/dev-docs

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #122 
- Closes #333 

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass
- [ ] If the PR changes the participant-level and/or the dataset-level result file, the [`query-tool-results` files](https://github.com/neurobagel/neurobagel_examples/tree/main/query-tool-results) in the  [neurobagel_examples repo](https://github.com/neurobagel/neurobagel_examples/) have been regenerated

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: cypress/component/ResultCard.cy.tsx [hunk 1]
```diff
@@ -41,12 +41,13 @@ describe('ResultCard', () => {
...
     cy.get('[data-cy="card-some uuid"] button')
+      .eq(1)
       .should('contain', 'ASL')
-      .should('have.class', 'bg-zinc-800');
+      .should('have.css', 'background-color', 'rgb(113, 113, 122)');
     cy.get('[data-cy="card-some uuid"] button')
       .eq(2)
       .should('contain', 'DWI')
-      .should('have.class', 'bg-red-700');
+      .should('have.css', 'background-color', 'rgb(205, 92, 92)');
```
Comment (surchs) [inline cypress/component/ResultCard.cy.tsx:50]: :cherries: are you not using tailwind classes for color here any more?

File: src/App.tsx [hunk 2]
```diff
@@ -476,10 +482,10 @@ function App() {
         </>
       )}
 
-      <div>{enableChatbot && <ChatbotFeature setResult={setResult} />}</div>
+      {enableChatbot && <ChatbotFeature setResult={setResult} />}
 
-      <div className="grid grid-cols-4 gap-4">
-        <div>
+      <div className="flex flex-wrap gap-3">
+        <div className="min-w-[380px] max-w-sm flex-1">
```
Comment (surchs) [inline src/App.tsx:489]: why did you pick 380px here? If that's for a good reason, I'd add a comment to explain why

File: src/App.tsx [hunk 3]
```diff
@@ -508,8 +514,18 @@ function App() {
...
-          <ResultContainer response={sortedResults || null} />
+        <div
+          className={
+            loading
+              ? 'flex flex-1 animate-pulse items-center justify-center'
+              : 'min-w-[600px] flex-1'
+          }
+        >
+          {loading ? (
+            <img src={logo} alt="Logo" className="max-h-20 animate-bounce" />
```
Comment (surchs) [inline src/App.tsx:526]: neat

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -58,11 +59,7 @@ function Navbar({
     <Toolbar className="my-4" data-cy="navbar">
       <div className="flex w-full items-center justify-between">
         <div className="flex items-center">
-          <img
-            src="https://raw.githubusercontent.com/neurobagel/documentation/main/docs/imgs/logo/neurobagel_logo.png"
-            alt="Logo"
-            height="60"
-          />
```
Comment (surchs) [inline src/components/Navbar.tsx:65]: OHO, good catch!

File: tailwind.config.js [hunk 5]
```diff
@@ -1,14 +1,7 @@
...
   content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
-  theme: {
-    extend: {
-      gridTemplateRows: {
-        9: 'repeat(9, auto)',
-        10: 'repeat(10, auto)',
-      },
-    },
-  },
+  theme: {},
```
Comment (surchs) [inline tailwind.config.js]: is that needed at all if it is empty?

**Review by surchs — APPROVED** [review]
Looks good @rmanaem, thanks for the work on this PR!

:cook:

File: cypress/component/ResultCard.cy.tsx (hunk 1 above)
Comment (rmanaem) [reply cypress/component/ResultCard.cy.tsx:50]: No, switched to rgb in this PR.

File: src/App.tsx (hunk 2 above)
Comment (rmanaem) [reply src/App.tsx:489]: It's a rather arbitrary value which I came to as I played around with the screen size. 380 px is the smallest we can go before the first dropdown placeholder text starts to lose a letter or so.

**Pull request merged**